examples/circuit-relaying/src/helpers.js

```diff
@@ -53,22 +55,64 @@ module.exports = (ipfs, peersSet) => {
     const tags = Array.from(peersSet).map((p) => {
       return `<li>${p}</li>`
     })
-    $peers.innerHTML = tags
+    $peers.innerHTML = tags.join('')
   }
 
-  const updateAddrs = (addrs) => {
-    $addrs.innerHTML = `
-        <div>
-            <ul>
-                ${addrs.map((addr) => `<li>${addr.toString()}</li>`)}
-            <ul>
-        </div>`
+  const updateSwarmPeers = async (ipfs) => {
+    const peers = await ipfs.swarm.peers()
+
+    $pAddrs.innerHTML = peers.map(peer => `<li>${peer.peer}</li>`).join('')
+  }
+
+  const updateAddrs = async (ipfs) => {
+    const info = await ipfs.id()
+
+    // see which peers support the circuit relay protocol
+    const relayAddrs = []
+    const connections = ipfs.libp2p.connections
+    const peers = await ipfs.swarm.peers()
+
+    for (let i = 0; i < peers.length; i++) {
+      const {
+        peer: peerId
+      } = peers[i]
+
+      const cons = connections.get(peerId)
+
+      for (let j = 0; j < cons.length; j++) {
+        const con = cons[j]
+        const remoteAddr = con.remoteAddr.toString()
+
+        for (let k = 0; k < 5; k++) {
+          // protocols is undefined until the connection is negotiated so try a few times
+          const protocols = ipfs.libp2p.peerStore.protoBook.get(con.remotePeer)
+
+          if (!protocols) {
+            await delay(500)
+            continue
+          }
+
+          const isRelay = protocols.find(proto => proto.toString().includes('/circuit/relay'))
+
+          // only add addresses for peers that are relays and that we aren't already
+          // connected to them via another relay
+          if (isRelay && !remoteAddr.includes('p2p-circuit') && !remoteAddr.includes(info.id)) {
+            relayAddrs.push(`${remoteAddr}/p2p-circuit/p2p/${info.id}`)
```

Comment:
A peer that supports the `/libp2p/circuit/relay/0.1.0` protocol isn't necessarily a relay and in the majority of cases won't be one.